auth/base/src/main/java/org/wildfly/security/auth/principal/NamePrincipal.java

```diff
@@ -93,4 +93,36 @@ public boolean equals(final NamePrincipal obj) {
     public String toString() {
         return name;
     }
+
+    /**
+     * Attempt to convert the given principal to a {@code NamePrincipal}.
+     *
+     * @param principal the original principal
+     * @return the {@code NamePrincipal} or {@code null} if the principal cannot be converted
+     */
+    public static NamePrincipal asNamePrincipal(Principal principal) {
+        if (principal instanceof NamePrincipal) {
+            return (NamePrincipal) principal;
+        }
+        return isNamePrincipal(principal) ? new NamePrincipal(principal.getName()) : null;
```

Comment:
We could always add later but if a NamePrincipal is as a result of conversion I wonder if we should consider somehow wrapping the original Principal - we probably can't consider the wrapped in terms of equality but maybe later if we had wrapped it it would give us an opportunity to unwrap.

I don't think this PR is the time for that but just a thought.

Reply:
Yeah, that's something that we should consider later. I've created https://issues.redhat.com/browse/ELY-2477 to track that.